expression/function_traits.go

```diff
@@ -42,6 +42,7 @@ var unFoldableFunctions = map[string]struct{}{
 	ast.GetVar:    {},
 	ast.GetParam:  {},
 	ast.Benchmark: {},
+	ast.DayName:   {},
```

Comment:
Why unfoldable?

Reply:
If we fold it, it's unable to check the function name in `BuildCastFunction`, since `expr` corresponding to `DayName` has been converted from `*ScalarFunction` to `*Constant`.